### ISSUE TITLE
confusing error message creating composer in invalid region

### ISSUE THREAD

**Issue by caoilte** [hashicorp/terraform-provider-google#10557]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

v1.0.10

### Affected Resource(s)

google_composer_environment

### Expected Behavior

When you attempt to create a cloud composer environment in a region it is not allowed (eg europe-west4) it would be very helpful to be informed of this. Failing this, some sort of troubleshooting help would short circuit narrowing down the issue.

### Actual Behavior

It just times out trying to create the environment, eg see gist
https://gist.github.com/caoilte/0071aaff3706abf35e28df073f83652a

**Comment by edwardmedia** [comment]
@caoilte I am with you but some are known issues. It is not something that can be implemented at the provider level. Did you notice below comment at its [doc](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/composer_environment). Does thie make sense?

> It can take up to one hour to create or update an environment resource. In addition, GCP may only detect some errors in configuration when they are used (e.g. ~40-50 minutes into the creation process), and is prone to limited error reporting. If you encounter confusing or uninformative errors, please verify your configuration is valid against GCP Cloud Composer before filing bugs against the Terraform provider. * Environments create Google Cloud Storage buckets that do not get cleaned up automatically on environment deletion. More about Composer's use of Cloud Storage. * Please review the known issues for Composer if you are having problems.

**Comment by caoilte** [comment]
@edwardmedia yup that makes sense. I did my best to validate my configuration, but there wasn't anything in the GCP docs or known issues to describe which regions the service is valid to create in. It wasn't until I literally tried using the GCP UI to create the cloud composer that I discovered it was region limited.

I don't know if that helps to clarify the docs or not, but feel free to close if you like.
